clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -317,4 +316,34 @@ public void testExpiredBatches() throws InterruptedException {
         List<RecordBatch> expiredBatches = accum.abortExpiredBatches(60, cluster, now);
         assertEquals(1, expiredBatches.size());
     }
+
+    @Test
+    public void testNonEmptyPartitionSendInProgress() throws InterruptedException {
+        long now = time.milliseconds();
```

Comment:
Also discussed offline with Becket, but this test can probably be simplified significantly